src/Illuminate/Database/Grammar.php

```diff
@@ -116,6 +116,62 @@ protected function wrapValue($value)
         return $value;
     }
 
+    /**
+     * Convert array into sql group.
+     *
+     * @param  mixed  $values
+     * @return string
+     */
+    public function group($values)
+    {
+        return '('.implode(', ', $values).')';
+    }
+
+    /**
+     * Recursively group arrays containing values into groups.
+     *
+     * @param  mixed  $values
+     * @return string
+     */
+    public function groupify($values)
+    {
+        if (is_array($values)) {
+            return $this->group(array_map([$this, __FUNCTION__], $values));
+        }
+
+        return $this->parameter($values);
+    }
+
+    /**
+     * Recursively group arrays containing columns into groups.
+     *
+     * @param  mixed  $values
+     * @return string
+     */
+    public function groupifyColumns($values)
+    {
+        if (is_array($values)) {
+            return $this->group(array_map([$this, __FUNCTION__], $values));
+        }
+
+        return $this->wrap($values);
+    }
+
+    /**
+     * Recursively group arrays containing raw values into groups.
+     *
+     * @param  mixed  $values
+     * @return string
+     */
+    public function groupifyRaw($values)
+    {
+        if (is_array($values)) {
+            return $this->group(array_map([$this, __FUNCTION__], $values));
```

Comment:
Is there any valid sql query where recursion is required? I would probably just pass `$values` without mapping.

Reply:
With raw we don't need to parse the value, so here it might not need to be recursive. But you can still have array in array so this just supports both single and 2d arrays